frontend/src/components/dashboard/aliases/CustomAddressGenerationModal.test.tsx

```diff
@@ -0,0 +1,266 @@
+  > = {
+    isOpen: true,
+    onClose,
+    onPick,
+    onUpdate,
+    subdomain: "vanessa",
+    aliasGeneratedState: false,
+    findAliasDataFromPrefix: (_prefix: string) =>
+      ({}) as unknown as import("../../../hooks/api/aliases").AliasData,
+  };
+
+  const utils = render(
+    <CustomAddressGenerationModal {...defaultProps} {...overrides} />,
+  );
+  return { ...utils, onClose, onPick, onUpdate };
+}
+
+beforeEach(() => {
+  mockUseProfiles.mockReturnValue({
+    data: [{ subdomain: "vanessa" } as ProfileData],
+  });
+});
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe("CustomAddressGenerationModal - creator flow", () => {
+  test("renders creator view with disabled submit when invalid and enables when valid", async () => {
+    const user = userEvent.setup();
+    const { onPick } = renderModal({ aliasGeneratedState: false });
+
+    expect(
+      screen.getByRole("heading", {
+        level: 3,
+        name: "modal-custom-alias-picker-heading-2",
+      }),
+    ).toBeInTheDocument();
+
+    const input = screen.getByPlaceholderText(
+      "modal-custom-alias-picker-form-prefix-placeholder-redesign",
+    ) as HTMLInputElement;
+
+    await user.type(input, "Bad!");
+    const submit = screen.getByRole("button", {
+      name: "modal-custom-alias-picker-form-submit-label-2",
+    });
+    expect(submit).toBeDisabled();
+
+    await user.click(submit);
+    expect(
+      screen.getByText("error-alias-picker-prefix-invalid"),
+    ).toBeInTheDocument();
+
+    await user.clear(input);
+    await user.type(input, "myMask");
+    expect(submit).toBeDisabled();
+
+    await user.clear(input);
+    await user.type(input, "mymask");
+    expect(submit).toBeEnabled();
+
+    await user.click(submit);
+    expect(onPick).toHaveBeenCalledTimes(1);
+    const [addressArg, setErrorStateArg] = onPick.mock.calls[0];
+    expect(addressArg).toBe("mymask");
+    expect(typeof setErrorStateArg).toBe("function");
+  });
+
+  test("cancel button triggers onClose", async () => {
+    const user = userEvent.setup();
+    const { onClose } = renderModal({ aliasGeneratedState: false });
+
+    const cancel = screen.getByRole("button", { name: "profile-label-cancel" });
+    await user.click(cancel);
+    expect(onClose).toHaveBeenCalledTimes(1);
+  });
+
+  test("shows composed domain hint using profile subdomain and runtime mozmailDomain", () => {
+    renderModal({ aliasGeneratedState: false });
+    expect(screen.getByText("@vanessa.mozmail.com")).toBeInTheDocument();
+  });
+});
+
+describe("CustomAddressGenerationModal - success flow", () => {
+  test("displays newly created mask and handles 'Copy mask' (copyToClipboard=true)", async () => {
+    const user = userEvent.setup();
+
+    const onUpdate = jest.fn();
+    const findAliasDataFromPrefix = jest
+      .fn()
+      .mockImplementation(
+        (_prefix: string) =>
+          ({}) as unknown as import("../../../hooks/api/aliases").AliasData,
```

Comment:
question (blocking): This module has `import { AliasData } from "../../hooks/api/aliases";` at the top already. Why do we need to re-import it from this separate path here?